GTG/core/dates.py

```diff
@@ -478,13 +477,20 @@ def _parse_only_month_day_for_recurrency(self, string, newtask=True):
         except ValueError:
             result = None
 
-        if result is None or result <= self_date:
+        # If it's not possible to recreate the date by replacing or
+        # newtask=False and the result is still less than self_date,
+        # we need to jump to the next month.
```

Comment:
You're just repeating the statement, you are missing the important WHY you are doing that. Maybe:

```suggestion
        # When creating a new task, make sure to use the current month instead
        # of the next month (when today_date <= mday), so that when creating a
        # recurring task the user should do the task "now".
        # (This also causes to be start_date > due_date)
```

(I also tried removing `not newtask`, and I think the behaviour is more reasonable with `result is None or result <= self_date`, so that when creating a recurring monthly task, if it is a previous day then don't do it this month, otherwise if today or later (or the start date or later) do it this month).

Reply:
>I think the behaviour is more reasonable with result is None or result < self_date

I've also tried it before, but when I tested it made other cases not work correctly.

Reply:
I did noticed later that `result < self_date` is buggy in case `result == self_date` when checking the latter date would result in an infinite loop, but I still think this behaviour can be made better (with proper reasoning in the comments)

Reply:
I've made a new PR that rewrites the function. let me know what you think. :)